### PR TITLE
Remove pa_lwt

### PR DESCRIPTION
**Pull request by rgrinberg**
Remove usage of the camlp4 lwt extension. Addressing part of #180.

### CODE REVIEW

File: lwt/cohttp_lwt_unix_io.ml [hunk 1]
```diff
@@ -30,36 +30,42 @@ let iter fn x = Lwt_list.iter_s fn x
...
+    >>= fun buf ->
+    CD.debug_print "<<<[%d] %s" count buf;
+    return buf
+  else
+    try_read ()
 
 let read_exactly ic buf off len =
+  let try_read () =
+    Lwt.try_bind (fun () -> Lwt_io.read_into_exactly ic buf off len)
+      (fun () -> return true)
```
Comment (avsm) [inline lwt/cohttp_lwt_unix_io.ml]: `return true` -> `Lwt.return_true`

File: lwt/cohttp_lwt_unix_io.ml [hunk 2]
```diff
@@ -30,36 +30,42 @@ let iter fn x = Lwt_list.iter_s fn x
...
+      (function
+        | End_of_file -> return false
```
Comment (avsm) [inline lwt/cohttp_lwt_unix_io.ml]: same for `Lwt.return_false`

File: lwt/cohttp_lwt_unix_net.ml [hunk 3]
```diff
@@ -43,15 +43,10 @@ let connect_uri ~ctx uri =
   >>= fun client ->
   Conduit_lwt_unix.connect ~ctx:ctx.ctx client
 
-let close_in ic =
-  ignore_result (try_lwt Lwt_io.close ic with _ -> return_unit)
+let close c = Lwt.catch (fun () -> Lwt_io.close c) (fun _ -> return_unit)
```
Comment (avsm) [inline lwt/cohttp_lwt_unix_net.ml]: have we lost `close'` here?

**Comment by avsm** [comment]
Definite yes from me!

File: lwt/cohttp_lwt_unix_net.ml (hunk 3 above)
Comment (rgrinberg) [inline lwt/cohttp_lwt_unix_net.ml]: It seems like `close'` was just a helper that was never exposed. It's no longer necessary.

**Comment by rgrinberg** [comment]
@avsm done.

**Pull request merged**